### PR TITLE
Added `{folder_id}` variable and parameter for subfolder support in `exp:file:entries` tag

### PR DESCRIPTION
**Pull request by intoeetive**
Code: https://github.com/ExpressionEngine/ExpressionEngine/pull/3823

### CODE REVIEW

**Review by jcogs-design — APPROVED** [review]
Works perfectly - thanks!

File: docs/add-ons/file.md [hunk 1]
```diff
@@ -219,6 +225,10 @@ If you have defined any [image manipulations](control-panel/file-manager/upload-
 
 The URL to the file.
 
+### `{folder_id}`
+
+If the file is in a subfolder, will return ID of the folder. If the file is in root or upload directory, will return 0.
```
Comment (TomJaeger) [inline docs/add-ons/file.md:230]: @intoeetive Will return the ID of the Subfolder?

File: docs/fieldtypes/file.md [hunk 2]
```diff
@@ -126,6 +126,10 @@ If you have defined any [image manipulations](control-panel/file-manager/upload-
     {file_size:small:human}
     {file_size:small:human_long}
 
+### `{folder_id}`
+
+If the file is in a subfolder, will return ID of the folder. If the file is in root or upload directory, will return 0.
```
Comment (TomJaeger) [inline docs/fieldtypes/file.md:131]: Same here.. will it return the ID of the subfolder?

**Review by TomJaeger — COMMENTED** [review]
Looking for a quick clarification

**Comment by jcogs-design** [comment]
In use `{folder_id}` returns the ID of the folder, which is confusingly referred to by `directory_id` in the CP URL when viewing the folder.
So for this CP URL:
```/cp/files/directory/20&directory_id=434```
`{directory_id}` returns `20`
`{folder_id}` returns `434`

In my use case this structure to successfully return the content of a sub-folder:
```{exp:file:entries folder_id="{folder_id}"}```

HTH
🐾

**Comment by intoeetive** [comment]
@TomJaeger folder and subfolder are really synonymes. Feel free to change if that does not sound clear

**Pull request merged**